### ISSUE TITLE
Troubles with dotnet pack

### ISSUE THREAD

**Issue by AndreyAkinshin** [dotnet/BenchmarkDotNet#375]
I'm trying to release `v0.10.3`, but `dotnet pack` print some warnings:
```
λ dotnet pack .\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj --configuration Release
Microsoft (R) Build Engine version 15.1.545.13942
Copyright (C) Microsoft Corporation. All rights reserved.

  BenchmarkDotNet.Core -> W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\_out\BenchmarkDotNet.Core.dll
  BenchmarkDotNet.Core -> W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\_out\BenchmarkDotNet.Core.dll
  Successfully created package 'W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\bin\Release\BenchmarkDotNet.Core.0.10.3.nupkg'.
C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(104,5): warning : Issue found with package 'BenchmarkDotNet.Core'. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(104,5): warning : Issue: Assembly not inside a framework folder. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(104,5): warning : Description: The assembly 'lib\BenchmarkDotNet.Core.dll' is placed directly under 'lib' folder. It is recommended that assemblies be placed inside a framework-specific folder. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(104,5): warning : Solution: Move it into a framework-specific folder. If this assembly is targeted for multiple frameworks, ignore this warning. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
```
Also, I don't have the `net4.5` and `netcoreapp1.1` subfolders inside the package:
![image](https://cloud.githubusercontent.com/assets/2259237/23392387/0dc39de0-fd9d-11e6-8207-c97499e576f4.png)

**Comment by adamsitnik** [comment]
It's weird but it works for me and the appveyor. I suspect that you changed the version, then called dotnet pack and it did not rebuild the dlls properly.

@AndreyAkinshin could you try the following steps:
```
git clean -xfd
dotnet restore
set the version in the common.props file
dotnet build .\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj -c Release
dotnet build .\src\BenchmarkDotNet.Toolchains.Roslyn\BenchmarkDotNet.Toolchains.Roslyn.csproj -c Release
dotnet build .\src\BenchmarkDotNet\BenchmarkDotNet.csproj -c Release
dotnet build .\src\BenchmarkDotNet.Diagnostics.Windows\BenchmarkDotNet.Diagnostics.Windows.csproj  -c Release
dotnet pack .\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj -c Release
dotnet pack .\src\BenchmarkDotNet.Toolchains.Roslyn\BenchmarkDotNet.Toolchains.Roslyn.csproj -c Release
dotnet pack .\src\BenchmarkDotNet\BenchmarkDotNet.csproj -c Release
dotnet pack .\src\BenchmarkDotNet.Diagnostics.Windows\BenchmarkDotNet.Diagnostics.Windows.csproj  -c Release
```

**Comment by AndreyAkinshin** [comment]
It doesn't help. =(
```
W:\Work\BenchmarkDotNet (master)
λ git clean -xfd

W:\Work\BenchmarkDotNet (master)
λ dotnet restore
  Restoring packages for W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.IntegrationTests\BenchmarkDotNet.IntegrationTests.csproj...
  Restoring packages for W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Toolchains.Roslyn\BenchmarkDotNet.Toolchains.Roslyn.csproj...
  Restoring packages for W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj...
  Restoring packages for W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Diagnostics.Windows\BenchmarkDotNet.Diagnostics.Windows.csproj...
  Restoring packages for W:\Work\BenchmarkDotNet\samples\BenchmarkDotNet.Samples\BenchmarkDotNet.Samples.csproj...
  Restoring packages for W:\Work\BenchmarkDotNet\src\BenchmarkDotNet\BenchmarkDotNet.csproj...
  Restoring packages for W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.Tests\BenchmarkDotNet.Tests.csproj...
  Generating MSBuild file W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Toolchains.Roslyn\obj\BenchmarkDotNet.Toolchains.Roslyn.csproj.nuget.g.props.
  Generating MSBuild file W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Diagnostics.Windows\obj\BenchmarkDotNet.Diagnostics.Windows.csproj.nuget.g.props.
  Generating MSBuild file W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Toolchains.Roslyn\obj\BenchmarkDotNet.Toolchains.Roslyn.csproj.nuget.g.targets.
  Generating MSBuild file W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Diagnostics.Windows\obj\BenchmarkDotNet.Diagnostics.Windows.csproj.nuget.g.targets.
  Writing lock file to disk. Path: W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Diagnostics.Windows\obj\project.assets.json
  Writing lock file to disk. Path: W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Toolchains.Roslyn\obj\project.assets.json
  Restore completed in 281.79 ms for W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Diagnostics.Windows\BenchmarkDotNet.Diagnostics.Windows.csproj.
  Restore completed in 281.68 ms for W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Toolchains.Roslyn\BenchmarkDotNet.Toolchains.Roslyn.csproj.
  Generating MSBuild file W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\obj\BenchmarkDotNet.Core.csproj.nuget.g.props.
  Generating MSBuild file W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\obj\BenchmarkDotNet.Core.csproj.nuget.g.targets.
  Writing lock file to disk. Path: W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\obj\project.assets.json
  Generating MSBuild file W:\Work\BenchmarkDotNet\src\BenchmarkDotNet\obj\BenchmarkDotNet.csproj.nuget.g.props.
  Generating MSBuild file W:\Work\BenchmarkDotNet\src\BenchmarkDotNet\obj\BenchmarkDotNet.csproj.nuget.g.targets.
  Writing lock file to disk. Path: W:\Work\BenchmarkDotNet\src\BenchmarkDotNet\obj\project.assets.json
  Restore completed in 1.5 sec for W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj.
  Restore completed in 1.52 sec for W:\Work\BenchmarkDotNet\src\BenchmarkDotNet\BenchmarkDotNet.csproj.
  Generating MSBuild file W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.Tests\obj\BenchmarkDotNet.Tests.csproj.nuget.g.props.
  Generating MSBuild file W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.Tests\obj\BenchmarkDotNet.Tests.csproj.nuget.g.targets.
  Writing lock file to disk. Path: W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.Tests\obj\project.assets.json
  Generating MSBuild file W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.IntegrationTests\obj\BenchmarkDotNet.IntegrationTests.csproj.nuget.g.props.
  Generating MSBuild file W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.IntegrationTests\obj\BenchmarkDotNet.IntegrationTests.csproj.nuget.g.targets.
  Writing lock file to disk. Path: W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.IntegrationTests\obj\project.assets.json
  Restore completed in 1.99 sec for W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.IntegrationTests\BenchmarkDotNet.IntegrationTests.csproj.
  Restore completed in 1.97 sec for W:\Work\BenchmarkDotNet\tests\BenchmarkDotNet.Tests\BenchmarkDotNet.Tests.csproj.
  Generating MSBuild file W:\Work\BenchmarkDotNet\samples\BenchmarkDotNet.Samples\obj\BenchmarkDotNet.Samples.csproj.nuget.g.props.
  Generating MSBuild file W:\Work\BenchmarkDotNet\samples\BenchmarkDotNet.Samples\obj\BenchmarkDotNet.Samples.csproj.nuget.g.targets.
  Writing lock file to disk. Path: W:\Work\BenchmarkDotNet\samples\BenchmarkDotNet.Samples\obj\project.assets.json
  Restore completed in 3.51 sec for W:\Work\BenchmarkDotNet\samples\BenchmarkDotNet.Samples\BenchmarkDotNet.Samples.csproj.

  NuGet Config files used:
      W:\Work\BenchmarkDotNet\NuGet.Config
      C:\Users\akinshin\AppData\Roaming\NuGet\NuGet.Config
      C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

  Feeds used:
      https://api.nuget.org/v3/index.json
      https://www.myget.org/F/fsharp-daily/api/v3/index.json
      https://dotnet.myget.org/F/dotnet-cli/api/v3/index.json
      https://dotnet.myget.org/F/dotnet-core/api/v3/index.json
      https://www.myget.org/F/xunit/api/v3/index.json

W:\Work\BenchmarkDotNet (master)
λ dotnet build .\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj -c Release
Microsoft (R) Build Engine version 15.1.545.13942
Copyright (C) Microsoft Corporation. All rights reserved.

  BenchmarkDotNet.Core -> W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\_out\BenchmarkDotNet.Core.dll
  BenchmarkDotNet.Core -> W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\_out\BenchmarkDotNet.Core.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:05.38

W:\Work\BenchmarkDotNet (master)
λ dotnet pack .\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj -c Release
Microsoft (R) Build Engine version 15.1.545.13942
Copyright (C) Microsoft Corporation. All rights reserved.

  BenchmarkDotNet.Core -> W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\_out\BenchmarkDotNet.Core.dll
  BenchmarkDotNet.Core -> W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\_out\BenchmarkDotNet.Core.dll
  Successfully created package 'W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\bin\Release\BenchmarkDotNet.Core.0.10.3.nupkg'.
C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(104,5): warning : Issue found with package 'BenchmarkDotNet.Core'. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(104,5): warning : Issue: Assembly not inside a framework folder. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(104,5): warning : Description: The assembly 'lib\BenchmarkDotNet.Core.dll' is placed directly under 'lib' folder. It is recommended that assemblies be placed inside a framework-specific folder. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(104,5): warning : Solution: Move it into a framework-specific folder. If this assembly is targeted for multiple frameworks, ignore this warning. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
```

**Comment by adamsitnik** [comment]
hmm @AndreyAkinshin let's try with `-v detailed` to get some more insight.

[My output](https://gist.github.com/adamsitnik/4bddece1b672964c0769751dd4f73323)

**Comment by AndreyAkinshin** [comment]
https://gist.github.com/adamsitnik/4bddece1b672964c0769751dd4f73323

I noticed that you have `Microsoft (R) Build Engine version 15.2.2.21747` and my version is `15.1.545.13942`. Could it be a problem? I used standard VS2017 RC4 isntaller. Where can I find an updated version?

**Comment by adamsitnik** [comment]
@AndreyAkinshin you posted url to my gist

**Comment by AndreyAkinshin** [comment]
Sorry, the correct link: https://gist.github.com/AndreyAkinshin/057384180ae0ceb12cfc80beee3648d5

**Comment by adamsitnik** [comment]
I can see that we have different `dotnet --version` Maybe you could try this one https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-dev-win-x64.latest.exe ? I have installed it yesterday when I was getting .NET Core 2.0 working on my pc according to [this instructions](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/dogfooding.md#how-to-get-up-and-running-on-net-core-20)

**Comment by adamsitnik** [comment]
btw @AndreyAkinshin the gist you provided seems to be only a part of the whole output

**Comment by AndreyAkinshin** [comment]
Look at this file: [out.txt](https://github.com/dotnet/BenchmarkDotNet/files/806098/out.txt)
For some reason, gist don't want to accept file with ~2200 lines. =(

**Comment by adamsitnik** [comment]
@AndreyAkinshin I can't see any significant difference. My only hope is installation of https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-dev-win-x64.latest.exe

**Comment by AndreyAkinshin** [comment]
It also didn't help. =(
```
W:\Work\BenchmarkDotNet (master)
λ dotnet --version
2.0.0-alpha-004853

W:\Work\BenchmarkDotNet (master)
λ dotnet pack .\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj -c Release
Microsoft (R) Build Engine version 15.2.2.21747
Copyright (C) Microsoft Corporation. All rights reserved.

  BenchmarkDotNet.Core -> W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\_out\BenchmarkDotNet.Core.dll
  BenchmarkDotNet.Core -> W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\_out\BenchmarkDotNet.Core.dll
C:\Program Files\dotnet\sdk\2.0.0-alpha-004853\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(89,5): warning : Issue found with package 'BenchmarkDotNet.Core'. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\2.0.0-alpha-004853\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(89,5): warning : Issue: Assembly not inside a framework folder. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\2.0.0-alpha-004853\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(89,5): warning : Description: The assembly 'lib\BenchmarkDotNet.Core.dll' is placed directly under 'lib' folder. It is recommended that assemblies be placed inside a framework-specific folder. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
C:\Program Files\dotnet\sdk\2.0.0-alpha-004853\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(89,5): warning : Solution: Move it into a framework-specific folder. If this assembly is targeted for multiple frameworks, ignore this warning. [W:\Work\BenchmarkDotNet\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj]
```
@adamsitnik, could you also share your output of  `dotnet build .\src\BenchmarkDotNet.Core\BenchmarkDotNet.Core.csproj -c Release`?

**Comment by AndreyAkinshin** [comment]
I found the problem! In my %PATH%, I had `C:\Program Files\dotnet\` and `C:\Program Files (x86)\dotnet\`. The x86 version was obsolete `1.0.0-preview2-1-003177`. Priority of the x64 version was higher, so `dotnet` was taken from the `x64` folder. But it seems that it take some information from the x86 folder (magic).

Anyway, I removed the x86 version from %PATH%, and now it works! It's hard to live on a bleeding edge. =(
I'm going to check final nupkg in all my environments and publish `v0.10.3` tonight.
